sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3460,15 +3462,23 @@ def _execute(
         Returns:
             Job id if the task is submitted to the cluster, None otherwise.
         """
-        if task.run is None:
+        if task.run is None and self._setup_cmd is None:
             logger.info('Run commands not specified or empty.')
```

Comment:
```suggestion
            logger.info('Both run and setup commands not specified or empty.')
```

Reply:
Actually, it can be a bit confusing to show the new message here. Reason: there are several cases without run section specified:
1. A user specifies the setup command without `--detach-setup`, the setup will be executed correctly and the original message should be good to shown.
2. A user specifies setup command with `--detach-setup`, the setup will be executed  in detached mode without showing this message.
3. A user does not specify setup or run, the original message seems still fine.

In that case, how about we keep the original message?

Reply:
Added some comments in the code : )

Reply:
An overall suggestion of adding too many comments: code should be self-explanatory and comments should be concise. For this specific case, I think it's fine to explain the cases when `self._setup_cmd is None` since it's less obvious (not just when there's no setup), but there's no need to explain why we don't write the code in certain way; that can further confuse the code reader